index.html

```diff
@@ -847,90 +847,77 @@ <h2>Examples</h2>
       </p>
 
       <pre class="example highlight" title="Create a new blank exposed Thing">
-        let thing = WoT.produce(thingDescription);
         try {
-          // manually add properties, actions, and events
+          var thing = WoT.produce({ name: "tempSensor" });
+          // manually add Interactions
           thing.addProperty({
             name: "temperature",
-            value: "0",
-            schema: '{ \"type\": \"number\" }',
-            writable: false
+            value: 0.0,
+            schema: '{ "type": "number" }'
             // use default values for the rest
-          });
-          thing.addEvent({
-            name: "onchange",
-            schema: '{ \"type\": \"number\" }'
-          });
-          thing.addAction({
+          }).addProperty({
+            name: "max",
+            value: 0.0,
+            schema: '{ "type": "number" }'
+            // use default values for the rest
+          }).addAction({
             name: "reset",
-            action: () => { this.temperature = 0; }
+            // no input, no output
+          }).addEvent({
+            name: "onchange",
+            schema: '{ "type": "number" }'
           });
           // add server functionality
-          thing.setPropertyReadHandler(propertyName => {
-            console.log("Handling read request for " + propertyName);
-            return new Promise((resolve, reject) => {
-                if (this.hasOwnProperty(propertyName))
-                  resolve(this[propertyName]);
-                else
-                  reject('Property not found');
-              },
-              e => {
-                console.log("Error");
-              });
-          });
+          thing.setActionHandler( () => {
+            console.log("Resetting maximum");
+            thing.writeProperty("max", 0.0);
+          }, "reset");
+
+          thing.start();
+
+          setInterval( async () => {
+            let mock = Math.random()*100;
+            thing.writeProperty("temperature", mock);
+            let old = await thing.readProperty("max");
+            if (old < mock) {
+              thing.writeProperty("max", mock);
+              thing.emitEvent("onchange");
+            }
+          }, 1000);
+
         } catch (err) {
-           console.log("Thing creation error.");
+           console.log("Script error: " + err);
         }
-        thing.start()
-        .then (() => {
-          thing.register();
-        });
       </pre>
```

Comment:
You removed the `start()` and `register()` part. It is important the reader understands that producing a Thing does not mean the Thing is running or is discoverable.

Reply:
Yes forgot. (Mainly because there is no real use case behind this.)

Reply:
Actually, there is a `thing.start()` further up in the example (before setting the interval to act upon the Thing).

Is `register()` also mandatory to call?